lib/dal/src/management/prototype.rs

```diff
@@ -486,25 +486,7 @@ impl ManagementPrototype {
             ManagerComponent::new(ctx, manager_component_id, &this_schema, &views).await?;
         let manager_component_geometry = manager_component.component.geometry.to_owned();
 
-        let mut variant_socket_map = HashMap::new();
-        let schemas = Schema::list(ctx).await?;
-        for schema in schemas {
-            let variant_id = Schema::default_variant_id_opt(ctx, schema.id()).await?;
-            if let Some(variant_id) = variant_id {
-                let (output_sockets, input_sockets) =
-                    SchemaVariant::list_all_socket_ids(ctx, variant_id).await?;
-
-                let has_manage_prototype =
-                    ManagementPrototype::variant_has_management_prototype(ctx, variant_id).await?;
-                let manage_count = if has_manage_prototype { 1 } else { 0 };
-
-                // 1 + for the Manager input socket
-                variant_socket_map.insert(
-                    schema.name().to_string(),
-                    1 + manage_count + output_sockets.len() + input_sockets.len(),
-                );
-            }
-        }
+        let variant_socket_map = Self::variant_socket_map(ctx).await?;
```

Comment:
execute_by_id() is getting rather large, so I moved variant_socket_map() into its own method rather than make it bigger.